plugins/thunderbolt/fu-thunderbolt-image.c

```diff
@@ -583,3 +585,18 @@ fu_plugin_thunderbolt_validate_image (GBytes  *controller_fw,
 
 	return hw_info->id != 0 ? VALIDATION_PASSED : UNKNOWN_DEVICE;
 }
+
+gboolean
+fu_plugin_thunderbolt_controller_is_native (GBytes    *controller_fw,
+					    gboolean  *is_native,
+					    GError   **error)
+{
+	const guint32 controller_sections[SECTION_COUNT] = { [DIGITAL_SECTION] = 0 };
+	gsize fw_size;
+	const guint8 *fw_data = g_bytes_get_data (controller_fw, &fw_size);
+	const FuThunderboltFwObject controller = { fw_data,   fw_size,   controller_sections };
+
+	const FuThunderboltFwLocation location = { .offset = 0x7B,  .len = 1, .description = "Native", .mask = 0x20 };
```

Comment:
Rather than hardcode this is multiple locations could you use get_host_location to get it?

Reply:
And then search the array for the one with description "native"?

The other option is to split this to a new function but then the image validation function will have to call it and compare this location specifically instead of having it together with the rest.

Reply:
Yeah I was thinking just search the array for native. If the offset changes in some new silicon rev or something it can abstract it better with needing to change less.